### PR TITLE
CRUD with BDBA Helm Chart

### PR DESCRIPTION
**Pull request by mphammer**

### CODE REVIEW

**Comment by operator-r2d2** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/1262#" title="Author self-approved">mphammer</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=blackducksoftware%2Fsynopsys-operator).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/blackducksoftware/synopsys-operator/blob/master/OWNERS)~~ [mphammer]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by operator-r2d2** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/1262#" title="Author self-approved">mphammer</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=blackducksoftware%2Fsynopsys-operator).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/blackducksoftware/synopsys-operator/blob/develop/OWNERS)~~ [mphammer]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: pkg/bdba/creater.go [hunk 1]
```diff
@@ -0,0 +1,79 @@
+/*
```
Comment (mphammer) [inline pkg/bdba/creater.go:1]: I will remove this file.

File: pkg/bdba/creater.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+	"encoding/json"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// GetPolarisReportingSecrets returns the image pull secret for Polaris Reporting
+// TODO: Refactor the Polaris code to use the same functions as Polaris-Reporting (this has less secrets and the secrets have different names)
+func GetPolarisReportingSecrets(namespace, GCPServiceAccount string) (map[string]runtime.Object, error) {
```
Comment (msenmurugan) [inline pkg/bdba/creater.go:34]: can we remove this method?

File: pkg/bdba/types.go [hunk 3]
```diff
@@ -0,0 +1,52 @@
+/*
```
Comment (mphammer) [inline pkg/bdba/types.go:1]: I think this file can be removed too.

File: pkg/bdba/ctl_bdba.go [hunk 4]
```diff
@@ -0,0 +1,514 @@
...
+		args:     map[string]interface{}{},
+		flagTree: FlagTree{},
+	}
+}
+
+// GetArgs returns the map of helm chart fields to values
+func (ctl *HelmValuesFromCobraFlags) GetArgs() map[string]interface{} {
+	return ctl.args
+}
+
```
Comment (msenmurugan) [inline pkg/bdba/ctl_bdba.go:150]: SetArgs method is missing which I have added recently to take care of upgrade scenario

File: pkg/synopsysctl/cmd_update.go [hunk 5]
```diff
@@ -1394,13 +1396,58 @@ var updatePolarisReportingCmd = &cobra.Command{
...
+	Args: func(cmd *cobra.Command, args []string) error {
+		// Check the Number of Arguments
+		if len(args) != 0 {
+			cmd.Help()
+			return fmt.Errorf("this command takes 0 argument, but got %+v", args)
+		}
+		return nil
+	},
+	RunE: func(cmd *cobra.Command, args []string) error {
+		// Get the flags to set Helm values
```
Comment (msenmurugan) [inline pkg/synopsysctl/cmd_update.go:1415]: add Get method to get the previous stored value and merge those values with new values

**Comment by operator-r2d2** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/1262#" title="Author self-approved">mphammer</a>*, *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/1262#pullrequestreview-386563455" title="Approved">msenmurugan</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=blackducksoftware%2Fsynopsys-operator).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/blackducksoftware/synopsys-operator/blob/develop/OWNERS)~~ [mphammer,msenmurugan]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by operator-r2d2** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/1262#" title="Author self-approved">mphammer</a>*, *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/1262#issuecomment-607934804" title="LGTM">msenmurugan</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=blackducksoftware%2Fsynopsys-operator).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/blackducksoftware/synopsys-operator/blob/develop/OWNERS)~~ [mphammer,msenmurugan]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**